here-naksha-lib-base/src/jvmMain/kotlin/naksha/base/Platform.kt

```diff
@@ -504,19 +514,47 @@ actual class Platform {
          */
         @Suppress("UNCHECKED_CAST")
         @JvmStatic
-        actual fun <T : Proxy> proxy(pobject: PlatformObject, klass: KClass<T>, doNotOverride: Boolean): T {
+        actual fun <T : Proxy> proxy(
+            pobject: PlatformObject,
+            klass: KClass<T>,
+            doNotOverride: Boolean
+        ): T {
             require(pobject is JvmObject)
             val symbol = Symbols.of(klass)
             var proxy = pobject.getSymbol(symbol)
             if (proxy != null) {
                 if (klass.isInstance(proxy)) return proxy as T
                 if (doNotOverride) throw IllegalStateException("The symbol $symbol is already bound to incompatible type")
             }
-            proxy = klass.primaryConstructor!!.call()
+
+            val constructor: KFunction<T> = primaryNonArgConstructorOf(klass)
+                ?: firstNonArgConstructorOf(klass)
+                ?: throw IllegalArgumentException("Unable to find primary or non-arg constructor for class: ${klass.qualifiedName}")
+            proxy = constructor.call()
```

Comment:
I think, any constructor has does not require arguments is fine for us, we should always first try the primary constructor, but if that does not match, try to find any other constructor.
These checks and searching for the correct constructor can be time consuming, I would suggest to add a cache here, a concurrent hash-map, to cache the found constructors. Like:
```kotlin
var constructor = cache[klass]
if (constructor == null) {
  constructor = ... find the correct one
  cache[klass] = constructor
}
```
We do not need any synchronization here, because we can assume that the finding operation is idempotent, so even when multiple threads in parallel enter the search and override each other, the cache should work after that as intended.

Reply:
done